### DISCUSSION TITLE
Companion watch app, heart rates, GPS tracking, etc.

### DISCUSSION THREAD

**Discussion by 9a3eedi** [wger-project/wger#1536] [Ideas]
I realize that wger is still a very young project, but I really see a lot of potential, as fitness data is a very personal thing worth self-hosting. I am very interested in switching to it as it matures. I was wondering if the following ideas are within its scope. I am basically looking for a self-hosted alternative to Google fit and Samsung Health, which let you track your workouts via smartwatches. 

Specifically I'd like to see wger support
- A smartwatch app that can read heart rates, etc.
- Heart rate tracking for workouts
- Step counts
- GPS tracking for e.g. hiking workouts or walking
- Calories burnt (defined by the workout + heart rates + other information)

Of course all this needs to be developed, but would it be considered within wger's roadmap, or is it out of scope?

**Comment by rolandgeider** [comment]
Hi! there are some ideas for a smartwatch app, specially the gym mode screams for one, but currently the flutter support is very minimal. I'm also not sure how easy it would be to access the sensors and read the heart rate etc out. One workaround (which is also a requested feature) would be adding a sync to read the saved values from such services, but that obviously wouldn't remove the dependency on them. So basically, besides the GPS tracking which is a completely different beast, I'd say that all that is in principle all in scope

**Comment by valerogarte** [comment]
Do you have any news on this topic? I would find it very interesting.

**Reply by rolandgeider** [reply]
hi! There hasn't been any progress yet.

There is an issue about connecting to android health (and the apple equivalent), to import your data into the system. 
https://github.com/wger-project/flutter/issues/618

and one for a watch app:
https://github.com/wger-project/flutter/issues/262

There's a couple of other things we're working on now, but the sync with probably be the next big feature, so perhaps sometime next year?
